modules/pytket-pyquil/pytket/extensions/pyquil/pyquil_convert.py

```diff
@@ -299,39 +312,84 @@ def process_characterisation(qc: QuantumComputer) -> dict:
     :type qc: QuantumComputer
     :return: A dictionary containing Rigetti device characteristics
     """
-    specs = qc.device.get_specs()
-
-    coupling_map = [
-        [n, ni]
-        for n, neigh_dict in qc.qubit_topology().adjacency()
-        for ni, _ in neigh_dict.items()
-    ]
+    isa = qc.quantum_processor.to_compiler_isa()
+    coupling_map = [[int(i) for i in e.ids] for e in isa.edges.values()]
+
+    str_to_gate_1qb = {
+        "RX": {
+            "PI": OpType.X,
+            "PIHALF": OpType.V,
+            "-PIHALF": OpType.Vdg,
+            "-PI": OpType.X,
+            "ANY": OpType.Rx,
+        },
+        "RZ": {
+            "ANY": OpType.Rz,
+        },
+    }
+    str_to_gate_2qb = {"CZ": OpType.CZ, "XY": OpType.ISWAP}
 
-    link_errors: dict = defaultdict(dict)
-    node_errors: dict = defaultdict(dict)
+    link_errors: Dict[Tuple[Node, Node], Dict[OpType, float]] = defaultdict(dict)
+    node_errors: Dict[Node, Dict[OpType, float]] = defaultdict(dict)
     readout_errors: dict = {}
-    t1_times_dict = {}
-    t2_times_dict = {}
-
-    device_node_fidelities = specs.f1QRBs()  # type: ignore
-    device_link_fidelities = specs.fCZs()  # type: ignore
-    device_fROs = specs.fROs()  # type: ignore
-    device_t1s = specs.T1s()  # type: ignore
-    device_t2s = specs.T2s()  # type: ignore
-
-    for index in qc.qubits():
-        node = Node(index)
-        readout_errors[node] = 1.0 - cast(float, device_fROs[index])
-        t1_times_dict[index] = device_t1s[index]
-        t2_times_dict[index] = device_t2s[index]
-        node_errors[node].update(
-            {OpType.Rx: 1.0 - cast(float, device_node_fidelities[index])}
-        )
-        # Rigetti use frame changes for Rz, so they effectively have no error.
-        node_errors[node].update({OpType.Rz: 0.0})
-
-    for (a, b), fid in device_link_fidelities.items():
-        link_errors[(Node(a), Node(b))].update({OpType.CZ: 1.0 - cast(float, fid)})
+    # T1s and T2s are currently left empty
+    t1_times_dict: dict = {}
+    t2_times_dict: dict = {}
+
+    for q in isa.qubits.values():
+        node = Node(q.id)
+        for g in q.gates:
+            if g.fidelity is None:
+                g.fidelity = 1.0
+            print(g)
+            if isinstance(g, GateInfo) and g.operator in str_to_gate_1qb:
+                angle = _get_angle_type(g.parameters[0])
+                if angle is not None:
+                    try:
+                        optype = str_to_gate_1qb[g.operator][angle]
+                    except KeyError:
```

Comment:
In this case I think `optype` won't be set to anything so the below code will fail; is there anything we can do in that case or should we just raise an error here?

Reply:
good spot, it needs to skip the rest in that case.